### DISCUSSION TITLE
Action 'huawei_appgallery_connect' isn't known to support operating system 'android'.

### DISCUSSION THREAD

**Discussion by BryanLeong99** [codemagic-ci-cd/codemagic-docs#1984] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

1. Run `fastlane init` at the root directory (android folder). 
2. Add the Fastfile scripts as follow:
 

> lane :huawei do
>     gradle(
>       task: 'bundle', #apk & aab
>       # build_type: 'Release',
>       build_type: 'HmsRelease',
>       print_command: true,
>     )
> 
>     huawei_appgallery_connect(
>       client_id: "#{ENV["APP_GALLERY_CLIENT_ID"]}",
>       client_secret: "#{ENV["APP_GALLERY_CLIENT_SECRET"]}",
>       app_id: "#{ENV["APP_GALLERY_APP_ID"]}",
>       apk_path: "./app/build/outputs/bundle/hms/release/app-hms-release.aab",
> 
>       # Optional, Parameter beyond this are optional
>       is_aab: true,
>       submit_for_review: false,
>       delay_before_submit_for_review: 20,
>     )
>   end

3. Add environment group that stores the necessary variables needed by Fastlane.
4. At the root directory, run the following scripts:

>    fastlane add_plugin huawei_appgallery_connect
>         bundle install
>         bundle exec fastlane huawei

        
  

### Expected results

Expected to be a successful build and will be uploaded to Huawei AppGallery. 

### Actual results

Instead, the build was successful, but when it runs the action, "huawei_appgallery_connect", it throws this message and did not upload the app to Huawei AppGallery.

> Action 'huawei_appgallery_connect' isn't known to support operating system 'android'.

### Build id (optional)

64e310d80d91075a0eca699c

**Comment by himesh-cm** [comment]
HI @BryanLeong99,
Apologies for the delayed reply, 

Upon checking the build, I can see that you are using something called `HmsRelease` instead of the normal `Release`, can you please share the reason for the same?
Maybe consider updating Fastlane as well, as this could be helpful.

Meanwhile, I will check it on my end to see if I can reproduce it with the script you shared.

**Reply by himesh-cm** [reply]
Hi @BryanLeong99, Did you get a chance to debug the issue as per my last message?

**Reply by BryanLeong99** [reply]
Because we customise our project into Huawei Mobile Service and Google Mobile Service, it works for our local build. 

However, I managed to solve the problem by running `fastlane new_action` and adding `plugins_path = File.join(File.dirname(__FILE__), 'fastlane', 'Pluginfile')
eval_gemfile(plugins_path) if File.exist?(plugins_path)` in the Gemfile.

**Reply by himesh-cm** [reply]
Hi @BryanLeong99,
Thanks for sharing it with us.
While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!
